components/dashboard/dashboard.go

```diff
@@ -127,11 +129,21 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 	}
 
 	// Deploy odh-dashboard manifests
-	if platform == deploy.OpenDataHub || platform == "" {
-		if err = deploy.DeployManifestsFromPath(cli, owner, Path, dscispec.ApplicationsNamespace, ComponentName, enabled); err != nil {
+	switch {
+	case platform == deploy.OpenDataHub, platform == deploy.Unknown:
+		base := Path
+		if dscispec.ServiceMesh.ManagementState == operatorv1.Managed {
+			base = PathServiceMesh
+		}
+		if err = deploy.DeployManifestsFromPath(cli, owner, base, dscispec.ApplicationsNamespace, ComponentName, enabled); err != nil {
 			return err
 		}
-	} else if platform == deploy.SelfManagedRhods || platform == deploy.ManagedRhods {
+
+		if err := d.configureServiceMesh(cli, owner, dscispec); err != nil {
```

Comment:
i am trying to  understand here:
the real call of `deploy.DeployManifestsFromPath(cli, owner, PathODHProjectController...` is moved into `configureServiceMesh`() ?